### PR TITLE
feat(.github): support dynamic submodule detection

### PR DESCRIPTION
**Pull request by codyoss**
Today the sub-modules are are coded into a couple of places to
work with release-please. This change is the first of three to
make release-please automatically support newly added modules.

Also, the workflow step was refactored to be in Go which makes
the code easier to test and debug locally in my opinion.

### CODE REVIEW

File: internal/actions/cmd/changefinder/main.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+		rootDir = os.Args[1]
+	}
+	log.Printf("Root dir: %q", rootDir)
+	var modDirs []string
+	// Find all external modules
+	filepath.WalkDir(rootDir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.Name() == "go.mod" && !strings.Contains(path, "/internal/") {
```
Comment (tbpg) [inline internal/actions/cmd/changefinder/main.go:44]: Can check if `Name` is `internal` and return `SkipDir` if so.

File: internal/actions/go.mod [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module cloud.google.com/go/internal/actions
```
Comment (tbpg) [inline internal/actions/go.mod:1]: Why do we need a separate module for this?

Comment (codyoss) [reply internal/actions/go.mod:1]: Technically no, but if I want to keep using newer io.FS apis, yes.

File: internal/actions/cmd/changefinder/main.go (hunk 1 above)
Comment (codyoss) [reply internal/actions/cmd/changefinder/main.go]: Done

**Review by tbpg — APPROVED** [review]

**Pull request merged**